test/libsolidity/GasCosts.cpp

```diff
@@ -38,7 +38,8 @@ namespace solidity::frontend::test
 #define CHECK_DEPLOY_GAS(_gasNoOpt, _gasOpt, _evmVersion) \
 	do \
 	{ \
-		u256 metaCost = GasMeter::dataGas(m_compiler.cborMetadata(m_compiler.lastContractName()), true, _evmVersion); \
+		auto compiledContract = m_compiler.output().contract().value(); \
```

Comment:
this unnecessarily copies around compile contract instances. I see you implemented a signature

```cpp
std::optional<CompiledContract> contract(ContractName const& _name = {}) const;
```

Instead, I would implement it as `CompiledContract const* contract(ContractName const& _name = {}) const;`. Ultimately the compiled contract comes from the `m_sourceUnits` map so all you have to do is point to it if it exists, return `nullptr` otherwise. Then you can also write the following

```suggestion
		auto const* compiledContract = m_compiler.output().contract().value(); \
```

Reply:
Agree! Changed the implementation.